### PR TITLE
Printing unicode messages

### LINKED ISSUE #70: Terminal output with utf8
Decode utf-8 messages printed to stdout.  The current behavior is to print base64-encoded messages, which are hard to verify.

## Example
`mailmerge_template.txt` with unicode characters, including emoji.
```
TO: to@test.com
FROM: from@test.com

Laȝamon 😀 klâwen
```

`mailmerge_database.csv`:
```
email
myself@mydomain.com
```

`mailmerge_server.conf`:
```yaml
[smtp_server]
host = open-smtp.example.com
port = 25
```

Running mailmerge results in the following stdout:
```console
$ mailmerge
>>> message 0
TO: to@test.com
FROM: from@test.com
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: base64
Date: Wed, 18 Dec 2019 15:02:04 -0000

TGHInWFtb24ga2zDondlbg==

>>> sent message 0
>>> Limit was 1 messages.  To remove the limit, use the --no-limit option.
>>> This was a dry run.  To send messages, use the --no-dry-run option.
```

Ideally, the output would look like this:
```console
Running mailmerge results in the following stdout:
```console
$ mailmerge
>>> message 0
TO: to@test.com
FROM: from@test.com
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: base64
Date: Wed, 18 Dec 2019 15:02:04 -0000

Laȝamon 😀 klâwen

>>> sent message 0
>>> Limit was 1 messages.  To remove the limit, use the --no-limit option.
>>> This was a dry run.  To send messages, use the --no-dry-run option.
```

## Testing
- Update existing `test_main.py::test_stdout_utf8`
- Don't print attachment content, just filename.  This is already the case.
- Non-multipart message
- HTML and plaintext multipart
- UTF8 multipart message
- Attachment
- CC
- BCC

### PR DESCRIPTION
**Pull request by awdeorio**
Print decoded message text instead of base64-encoded text at the CLI.

Closes #70

### CODE REVIEW

**Comment by awdeorio** [comment]
@seshrs I could use some input on output that makes sense.  Here's my first stab at it.  Are the headers confusing or helpful?  There are no multipart boundaries, is that confusing?

```console
$ mailmerge --resume 2
>>> message 2
TO: Lazamon<two@test.com>
FROM: from@test.com
CC: cc1@test.com, cc2@test.com
MIME-Version: 1.0
Content-Type: multipart/alternative; boundary="boundary"
Date: Mon, 27 Jan 2020 20:32:08 -0000

MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: base64

Laȝamon 😀 klâwen



MIME-Version: 1.0
Content-Type: text/html; charset="utf-8"
Content-Transfer-Encoding: base64

<html>
  <body>
    <p>Laȝamon 😀 klâwen</p>
  </body>
</html>

Content-Type: application/octet-stream; Name="attachment1.txt"
MIME-Version: 1.0
Content-Transfer-Encoding: base64
Content-Disposition: attachment; filename="attachment1.txt"

Content-Type: application/octet-stream; Name="attachment2.csv"
MIME-Version: 1.0
Content-Transfer-Encoding: base64
Content-Disposition: attachment; filename="attachment2.csv"

>>> attached attachment1.txt
>>> attached attachment2.csv
>>> sent message 2
>>> Limit was 1 message.  To remove the limit, use the --no-limit option.
>>> This was a dry run.  To send messages, use the --no-dry-run option.
```

**Comment by seshrs** [comment]
I think this is already looking great!

Here are my opinions:

- For attachments, we already print `>>> attached attachment1.txt`. In this case, printing those attachment message headers is probably redundant, and can be removed from the output.
- If we agree about the above, then the printed message is really only an informative representation of the email anyway. Hence, it's probably better to not show the confusing boundaries. If it's not too much work, we could also leave out some of the headers in each part. (See suggested output below.)
- The rest of the output looks fine to me. Perhaps we can take this opportunity to also beautify the output with colors? I was thinking of something like this:

image

(I read a StackOverflow answer describing [Blessings](https://pypi.org/project/blessings/) + [Colorama](https://pypi.org/project/colorama/).)

What do you think?

**Comment by awdeorio** [comment]
I really like the color idea.  I'll give it a shot and post a revised example.

**Comment by awdeorio** [comment]
Here's the latest draft, LMK what you think.

Screen Shot 2020-01-29 at 10 42 08 AM

File: mailmerge/__main__.py [hunk 1]
```diff
@@ -101,11 +106,9 @@ def main(sample, dry_run, limit, no_limit, resume,
         for _, row in enumerate_range(csv_database, start, stop):
             sender, recipients, message = template_message.render(row)
             sendmail_client.sendmail(sender, recipients, message)
-            print(">>> message {}".format(message_num))
-            print(message.as_string())
-            for filename in get_attachment_filenames(message):
-                print(">>> attached {}".format(filename))
-            print(">>> sent message {}".format(message_num))
+            print(TERM.reverse_bold_cyan(">>> message {}".format(message_num)))
+            print_message(message)
```
Comment (seshrs) [inline mailmerge/__main__.py]: Perhaps these two print statements should appear before actually sending the message?

File: mailmerge/__main__.py [hunk 2]
```diff
@@ -269,20 +272,29 @@ def enumerate_range(iterable, start=0, stop=None):
...
+            charset = str(part.get_charset())
+            print(part.get_payload(decode=True).decode(charset))
+            print()
+        elif is_attachment(part):
+            print(TERM.cyan(">>> message part: attachment {}".format(part.get_filename())))
+        else:
+            print(TERM.cyan(">>> message part: {}".format(part.get_content_type())))
+
+
+def is_attachment(part):
```
Comment (seshrs) [inline mailmerge/__main__.py]: 👍

File: mailmerge/__main__.py (hunk 1 above)
Comment (awdeorio) [reply mailmerge/__main__.py]: Good call!  25c42cc998e6cfb31f22071e9ee25d8579de41e8

**Pull request merged**